File: Solution/0001.Two_Sum/0001.Two_Sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        d = dict()
        for i in range(len(nums)):
            anchor = target-nums[i]
            if d.has_key(anchor) == False:
                d[nums[i]]=i
            else:
                return [d[anchor],i]
        return []

File: Solution/0001.Two_Sum/0001.Two_Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int,int> m;
        for(int i=0;i<nums.size();++i)
        {
            int anchor=target-nums[i];    //trick 把要找的目标值记录下来，以免重复计算
            if(m.count(anchor)==0)
                m[nums[i]]=i;
            else
                return {m[anchor],i};   
        }
        return {};

    }
};
